internal/witness/verifier/verifier.go

```diff
@@ -26,21 +26,21 @@ import (
 )
 
 // WitnessV consists of a CT signature verifier.
-type WitnessV struct {
+type WitnessVerifier struct {
 	SigVerifier *ct.SignatureVerifier
 }
 
 // NewWitnessVerifier creates a witness signature verifier from a public key.
-func NewWitnessVerifier(pk crypto.PublicKey) (*WitnessV, error) {
+func NewWitnessVerifier(pk crypto.PublicKey) (*WitnessVerifier, error) {
 	sv, err := ct.NewSignatureVerifier(pk)
 	if err != nil {
 		return nil, fmt.Errorf("failed to create signature verifier: %v", err)
 	}
-	return &WitnessV{SigVerifier: sv}, nil
+	return &WitnessVerifier{SigVerifier: sv}, nil
 }
 
 // VerifySignature verifies the witness signature on a cosigned STH.
-func (wv WitnessV) VerifySignature(sth api.CosignedSTH) error {
+func (wv WitnessVerifier) VerifySignature(sth api.CosignedSTH) error {
 	if len(sth.WitnessSigs) == 0 {
```

Comment:
Just spotted that the code in here only verifies the _first_ witness signature - should probably change that and document the semantics for what partial success means?

Reply:
Yes, good point!  I just committed something that seems more reasonable - I'll plan to write a test too so let me know if it seems important to include that in this PR.  

I guess what we'd eventually want (but doesn't seem worth implementing now) is for `WitnessVerifier` to take in multiple underlying keys / verifiers and output all the signatures that verify, Note-style.

Reply:
Yeah, that seems like the right direction - may be worth popping a TODO() in there to just capture that for future us?

Reply:
I didn't put it as a TODO (because we might not end up doing it 😄) but I put a comment to the same effect.